modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -342,6 +342,7 @@ private MetadataAttributes getMetadataAttributes(MetadataAttributes.MetadataAttr
                                         parameterModel.getDslConfiguration(),
                                         parameterModel.getDisplayModel().orElse(null),
                                         parameterModel.getLayoutModel().orElse(null),
+                                        parameterModel.getValuesProviderModel().orElse(null),
```

Comment:
this is weird. Why should a MetadataAttributes know about this? Seems like two completely separate concerns to me.